### PR TITLE
Moving cassandra compatibility to Explore

### PR DESCRIPTION
**Pull request by premkumr**
-  Create a section in explore for YCQL
- Add Cassandra compatibility page

@netlify /preview/explore/ycql-language/cassandra-feature-support/

### CODE REVIEW

File: docs/content/preview/explore/ycql-language/_index.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+---
+YugabyteDB's [YCQL API](../../api/ycql/) has its roots in the [Cassandra Query Language (CQL)](http://cassandra.apache.org/doc/latest/cql/index.html) and runs on top of YugabyteDB's distributed storage layer called DocDB. This architecture allows YCQL to support most Cassandra features, such as data types, queries, expressions, operators and so on and at the same providing seamless scalability and reilience.
+
+{{< tip title="Tip" >}}
+A large portion of the documentation and examples written for Cassandra would work against YCQL.
+{{< /tip >}}
+
+## Cassandra features in YCQL
+
+The following table lists the most important YCQL features which you would find familiar if you have worked with PostgreYCQL.
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/_index.md:24]:
```suggestion
The following table lists the most important YCQL features which you would find familiar if you have worked with Cassandra.
```

File: docs/content/preview/explore/ycql-language/_index.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+---
+title: YCQL features
+headerTitle: YCQL features
+linkTitle: YCQL features
+description: Explore core features in YCQL
+headcontent: Explore core features in YCQL
+image: /images/section_icons/api/ycql.png
+menu:
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/_index.md:8]: Should it be 'CQL features' and 'Explore core CQL features in YCQL'?

Comment (premkumr) [reply docs/content/preview/explore/ycql-language/_index.md:8]: The section is going to be about `YCQL`. Only the `Cassandra feature support` page is about Cassandra ..

File: docs/content/preview/explore/ycql-language/_index.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+| [Security](../../api/ycql/#ddl-security-statements) | Roles and Permissions |
+
+## Going beyond Cassandra
+
+YCQL has a number of features that are not present in Cassandra, as summarized in the following table.
+
+| YCQL Feature | Description |
+| :----------- | :---------- |
+| [Strongly consistent with RAFT replication](../../architecture/docdb-replication/replication/#raft-replication) | Enables strong consistency across replicas |
+| [Fast transactions](../../architecture/transactions/distributed-txns/) | Single round-trip transactions |
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/_index.md:42]: Distributed transactions? (Cassandra would also be considered fast, no?)

File: docs/content/preview/explore/ycql-language/_index.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+| [Native JSONB support](going-beyond-YCQL/follower-reads-yYCQL/)| Enables document data modelling like MongoDB |
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/_index.md:43]: broken link

File: docs/content/preview/explore/ycql-language/cassandra-feature-support.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
+---
+title: Cassandra feature support
+linkTitle: Cassandra feature support
+description: Summary of YugabyteDB's parity to Cassandra features
+headcontent: YugabyteDB supports most standard Cassandra features.
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/cassandra-feature-support.md:5]:
```suggestion
headcontent: YugabyteDB supports most standard Cassandra features
```

File: docs/content/preview/explore/ycql-language/cassandra-feature-support.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+Yugabyte Cloud Query Language (YCQL) has its roots in the [Cassandra Query Language (CQL)](http://cassandra.apache.org/doc/latest/cql/index.html). This page highlights the important differences in feature support between YCQL and Cassandra 3.4.2.
+
+## Data types
+
+### Primitive Types
+
+|                |   Category   |                                            Types                                             |
+| :------------: | ------------ | :------------------------------------------------------------------------------------------- |
+| {{<icon/yes>}} | Integers     | [BIGINT , COUNTER, INT, INTEGER, SMALLINT, TINYINT, VARINT](../../../api/ycql/type_int)      |
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/cassandra-feature-support.md:27]:
```suggestion
| {{<icon/yes>}} | Integers     | [BIGINT, COUNTER, INT, INTEGER, SMALLINT, TINYINT, VARINT](../../../api/ycql/type_int)      |
```

File: docs/content/preview/explore/ycql-language/cassandra-feature-support.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+| {{<icon/yes>}} | UUID         | [TIMEUUID, UUID](../../../api/ycql/type_uuid)                                                |
+| {{<icon/no>}}  | TUPLE        |                                                                                              |
+{.sno-1}
+
+### User defined data types
+
+|                |    Operation    |                     Details                      |
+| :------------: | :-------------- | ------------------------------------------------ |
+| {{<icon/yes>}} | Create new type | [CREATE TYPE](../../../api/ycql/ddl_create_type) |
+| {{<icon/yes>}} | Delete types    | [DROP TYPE](../../../api/ycql/ddl_create_type)   |
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/cassandra-feature-support.md:45]:
```suggestion
| {{<icon/yes>}} | Delete types    | [DROP TYPE](../../../api/ycql/ddl_drop_type)   |
```

File: docs/content/preview/explore/ycql-language/cassandra-feature-support.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+### Select
+
+|                |                 Operation                 |                                 Details                                  |
+| :------------: | :---------------------------------------- | :----------------------------------------------------------------------- |
+| {{<icon/yes>}} | Select columns                            | [SELECT * FROM ...](../../../api/ycql/dml_select/)                       |
+| {{<icon/yes>}} | Conditional select with `[NOT] IN` clause | [SELECT ... WHERE key IN ...](../../../api/ycql/dml_select#where-clause) |
+| {{<icon/yes>}} | Conditional select with `IF` clause       | [SELECT ... IF ...](../../../api/ycql/dml_select#if-clause)              |
+| {{<icon/no>}}  | Select using `CONTAINS [KEY]`             | {{<issue 3620 "Tracked">}}                                               |
+| {{<icon/no>}}  | `SELECT JSON`                             | But [JSONB](../../../api/ycql/type_jsonb/) is supported as a native type |
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/cassandra-feature-support.md:94]:
```suggestion
| {{<icon/no>}}  | `SELECT JSON`                             | [JSONB](../../../api/ycql/type_jsonb/) is supported as a native type |
```

File: docs/content/preview/explore/ycql-language/cassandra-feature-support.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+| {{<icon/yes>}} | Delete with `USING` clause                | [DELETE ... USING](../../../api/ycql/dml_delete#using-clause)            |
+| {{<icon/no>}}  | Conditional Delete using `CONTAINS [KEY]` | `DELETE ... WHERE <col> CONTAINS ...`                                    |
+{.sno-1}
+
+### Insert
+
+|                |   Operation    |                                 Details                                  |
+| :------------: | :------------- | :----------------------------------------------------------------------- |
+| {{<icon/yes>}} | Adding columns | [INSERT ... INTO ...](../../../api/ycql/dml_insert/)                     |
+| {{<icon/no>}}  | `INSERT JSON`  | But [JSONB](../../../api/ycql/type_jsonb/) is supported as a native type |
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/cassandra-feature-support.md:126]:
```suggestion
| {{<icon/no>}}  | `INSERT JSON`  | [JSONB](../../../api/ycql/type_jsonb/) is supported as a native type |
```

File: docs/content/preview/explore/ycql-language/cassandra-feature-support.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+| {{<icon/yes>}} | SQL style transaction commit | [COMMIT](../../../api/ycql/dml_transaction#sql-syntax)            |
+{.sno-1}
+
+## Security
+
+|                |  Component   |                                    Details                                     |
+| :------------: | :----------- | :----------------------------------------------------------------------------- |
+| {{<icon/yes>}} | Roles        | [Manage users and roles](../../../secure/authorization/create-roles-ycql/)     |
+| {{<icon/yes>}} | Permissions  | [Grant privileges](../../../secure/authorization/ycql-grant-permissions/)      |
+| {{<icon/no>}}  | Users        | Legacy Cassandra feature (_CREATE, DROP, ALTER,LIST_)                          |
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/cassandra-feature-support.md:145]:
```suggestion
| {{<icon/no>}}  | Users        | Legacy Cassandra feature (_CREATE, DROP, ALTER, LIST_)                          |
```

File: docs/content/preview/explore/ycql-language/cassandra-feature-support.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+|   {{<icon/yes>}}   | Operators                    | [Binary, Unary, Null operators](../../../api/ycql/expr_ocall/)                 |
+| {{<icon/partial>}} | Batch                        | Only programattically via [BatchStatement](../../../api/ycql/batch/)           |
+|   {{<icon/no>}}    | Materialized Views           |                                                                                |
+|   {{<icon/no>}}    | Triggers                     |                                                                                |
+|   {{<icon/no>}}    | User Defined Aggregates(UDA) |                                                                                |
+|   {{<icon/no>}}    | User-defined functions(UDF)  |                                                                                |
+{.sno-1}
+
+## Learn more
+
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/cassandra-feature-support.md:165]: Both links broken

File: docs/assets/scss/_code_block.scss [hunk 12]
```diff
@@ -12,7 +12,7 @@ main {
       line-height: 110%;
       max-width: 100%;
       background: #f2f6ff;
-      border: 1px solid #cbdbff;
+      border: 2px solid #f2f6ff;
```
Comment (ddhodge) [inline docs/assets/scss/_code_block.scss:15]: Not sure why you removed border? Doesn't work so well in tabs for example 
https://deploy-preview-17906--infallible-bardeen-164bc9.netlify.app/preview/explore/#set-up-yugabytedb-universe  and scroll down a little

File: docs/content/preview/explore/ysql-language-features/sql-feature-support.md [hunk 13]
```diff
@@ -16,130 +16,138 @@ To understand which standard SQL features we support, refer to the following tab
```
Comment (ddhodge) [inline docs/content/preview/explore/ysql-language-features/sql-feature-support.md:16]: Replace sentence with "This page highlights the important differences in feature support between YSQL and SQL.

**Review by ddhodge — COMMENTED** [review]
Misc comments

File: docs/assets/scss/_code_block.scss (hunk 12 above)
Comment (premkumr) [reply docs/assets/scss/_code_block.scss:15]: code blocks looks cleaner, but on tabs we have to do something else. I'll retain this for simple back ticks. I'll revert the change for code blocks.

Comment (ddhodge) [reply docs/assets/scss/_code_block.scss:15]: You need to bump the border width back to 1 as well

Comment (ddhodge) [reply docs/assets/scss/_code_block.scss:15]: image

Not convinced...

Comment (premkumr) [reply docs/assets/scss/_code_block.scss:15]: Ok, let's discuss this separately. When simple ticks are used in a sentence, it flows well. But the tables are striped. I'll revert it for now.

**Comment by premkumr** [comment]
Let me know if this looks good now

File: docs/content/preview/explore/ycql-language/_index.md (hunk 2 above)
Comment (ddhodge) [reply docs/content/preview/explore/ycql-language/_index.md:8]: Right, but in Explore the intention is to introduce CQL users to YugabyteDB - they don't even know what YCQL is at this point, though they might guess. I thought the intention was to mirror the 'SQL features' section. That is also about YSQL basically, but 'SQL' orients them, no?

**Review by ddhodge — APPROVED** [review]
What about menu order, should it be at the bottom, or after SQL features?

**Comment by premkumr** [comment]
> What about menu order, should it be at the bottom, or after SQL features?

@ddhodge. We will have to add more pages to these sections & in that process will work on re-structuring and ordering.

**Pull request merged**